src/wallet/wallet2.cpp

```diff
@@ -8484,6 +8489,7 @@ void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>>
         {
           num_found = 1;
           seen_indices.emplace(td.m_global_output_index);
+          seen_outputs.emplace({amount, td.m_global_output_index});
```

Comment:
Adding to [this comment](https://github.com/monero-project/monero/pull/8047/files#r769864024), I think you need to emplace the real outputs in `seen_outputs` before iterating to select decoys (much earlier on in `get_outs`, otherwise if you have 2 rings, ring 1 would be able to include a decoy that is a real output from ring 2)